### PR TITLE
MDEXP-684: Avoid manual configuration for data-export

### PR DESCRIPTION
**Pull request by DmytroBykov1**
## Purpose
https://folio-org.atlassian.net/browse/MDEXP-684
Views migration should be reworked to skip env variable usage.

## Approach
Introduce setup of system var value (equal to tenant value) at the time of the general tenant upgrade flow

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] Check logging
  - [x] There are no major code smells or security issues

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,3 +144,21 @@ The [raml-module-builder](https://github.com/folio-org/raml-module-builder) fram
...
+
+| Name                    | Default value          | Description                                |
+|:------------------------|:-----------------------|:-------------------------------------------|
+| S3_URL                  | http://127.0.0.1:9000/ | S3 url                                     |
+| S3_REGION               | -                      | S3 region                                  |
+| S3_BUCKET               | -                      | S3 bucket                                  |
+| S3_ACCESS_KEY_ID        | -                      | S3 access key                              |
+| S3_SECRET_ACCESS_KEY    | -                      | S3 secret key                              |
+| S3_IS_AWS               | false                  | Specify if AWS S3 is used as files storage |
+| MY_UNIVERSITY_VALUE     | -                      | The tenant definition value                |
```
Comment (NikitaSedyx) [inline README.md:162]: not sure that this var should be documented as it's system and dynamic, there is no reason to set up it outside

File: src/main/java/org/folio/dataexp/service/DataExportTenantService.java [hunk 2]
```diff
@@ -57,6 +70,13 @@ public void loadReferenceData() {
     loadConfiguration();
   }
 
+  @Override
+  public synchronized void createOrUpdateTenant(TenantAttributes tenantAttributes) {
+      var tenantId = super.context.getTenantId();
+      var tenant = StringUtils.isEmpty(tenantId) ? myUniversity : tenantId;
+      System.setProperty(VIEWS_VARIABLE_FOR_SQL_SCRIPT, tenant);
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/dataexp/service/DataExportTenantService.java:77]: it should be always taken from request headers, no reason to use value defined outside

File: README.md (hunk 1 above)
Comment (DmytroBykov1) [reply README.md]: fixed

File: src/main/java/org/folio/dataexp/service/DataExportTenantService.java (hunk 2 above)
Comment (DmytroBykov1) [reply src/main/java/org/folio/dataexp/service/DataExportTenantService.java:64]: fixed

**Review by alekGbuz — APPROVED** [review]

File: src/main/java/org/folio/dataexp/service/DataExportTenantService.java [hunk 3]
```diff
@@ -33,11 +34,11 @@ public class DataExportTenantService extends TenantService {
 
   private static final List<String> JOB_PROFILES = List.of("default_authority_job_profile.json",
     "default_holdings_job_profile.json", "default_instance_job_profile.json");
+  public static final String VIEWS_VARIABLE_FOR_SQL_SCRIPT = "myuniversity";
```
[the remark this answers is not in the document]
Comment (DmytroBykov1) [reply src/main/java/org/folio/dataexp/service/DataExportTenantService.java]: done

File: src/main/java/org/folio/dataexp/service/DataExportTenantService.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public class DataExportTenantService extends TenantService {
 
   private static final List<String> JOB_PROFILES = List.of("default_authority_job_profile.json",
     "default_holdings_job_profile.json", "default_instance_job_profile.json");
-  public static final String VIEWS_VARIABLE_FOR_SQL_SCRIPT = "myuniversity";
+  public static final String COMMON_TENANT_FOR_VIEWS = "myuniversity";
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/dataexp/service/DataExportTenantService.java:37]: it's not common tenant

Comment (DmytroBykov1) [reply src/main/java/org/folio/dataexp/service/DataExportTenantService.java]: fixed

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**